emily/handler/src/common/error.rs

```diff
@@ -270,40 +270,29 @@ impl Error {
     /// Converts the error into a warp response.
     pub fn into_response(self) -> warp::reply::Response {
         warp::reply::with_status(
-            warp::reply::json(&ErrorResponse { message: format!("{self:?}") }),
+            warp::reply::json(&ErrorResponse { message: format!("{self}") }),
             self.status_code(),
         )
         .into_response()
     }
     /// Convert error into a presentable version of the error that can be
     /// provided to a client in production.
-    ///
-    /// TODO(131): Scrutinize the outputs of the error messages to ensure they're
-    /// production ready.
     pub fn into_production_error(self) -> Error {
         match self {
-            Error::Network(_)
-            | Error::VersionConflict(_)
-            | Error::Reorganizing(_)
-            | Error::Base64Decode(_)
-            | Error::EnvVariable(_)
-            | Error::SerdeJson(_)
-            | Error::SerdeDynamo(_)
-            | Error::EnvParseInt(_)
-            | Error::InvalidDepositEntry(_, _)
-            | Error::InvalidWithdrawalEntry(_, _)
-            | Error::AwsSdkDynamoDbDeleteItem(_)
-            | Error::AwsSdkDynamoDbGetItem(_)
-            | Error::AwsSdkDynamoDbPutItem(_)
-            | Error::AwsSdkDynamoDbQuery(_)
-            | Error::AwsSdkDynamoDbScan(_)
-            | Error::AwsSdkDynamoDbUpdateItem(_)
-            | Error::InternalServer => Error::InternalServer,
-            #[cfg(feature = "testing")]
-            Error::DynamoDbBuild(_) | Error::AwsSdkDynamoDbBatchWriteItem(_) => {
-                Error::InternalServer
-            }
-            err => err,
+            Error::DepositOutputMismatch(_, _)
+            | Error::Forbidden
+            | Error::NotFound
+            | Error::TooManyInternalRetries
+            | Error::InconsistentState(_)
+            | Error::Deserialization(_)
+            | Error::WithdrawalRequestIdMismatch(_, _)
+            | Error::InvalidStacksAddress(_)
```

Comment:
`InvalidStacksAddress` seems an env related error (coming from `Settings::from_env`), and although the to str isn't leaking anything maybe we don't want to expose it. WDYT?